integration/testrunner/http.go

```diff
@@ -64,18 +71,57 @@ func testHTTPEndpoint(ctx context.Context, tb testing.TB, endpointURL, idToken,
 	}); err != nil {
 		tb.Fatal(err)
 	}
-
 	bqClient := makeBigQueryClient(ctx, tb, projectID)
-
+	time.Sleep(10 * time.Second)
 	bqQuery := makeQueryForHTTP(*bqClient, id, projectID, datasetQuery)
-	queryIfAuditLogExistsWithRetries(ctx, tb, bqQuery, cfg, "httpEndpointTest")
+	tb.Log(bqQuery.Q)
+	results := queryIfAuditLogsExists(ctx, tb, bqQuery, cfg, "httpEndpointTest")
+	wantNum := 1
+	if len(results) != wantNum {
+		tb.Errorf("log number doesn't match (-want +got):\n - %d\n + %d\n", wantNum, len(results))
+	}
+	jsonPayloadInfo := parseJsonpayload(tb, results[0])
+	diff := diffResults(jsonPayloadInfo, getMode("HTTP"))
+	if diff != "" {
+		tb.Errorf(diff)
+	}
 }
 
 func makeQueryForHTTP(client bigquery.Client, id, projectID, datasetQuery string) *bigquery.Query {
-	// Cast to int64 because the result checker expects a number.
-	queryString := fmt.Sprintf("SELECT CAST(EXISTS (SELECT * FROM `%s.%s` WHERE labels.trace_id=?", projectID, datasetQuery)
-	queryString += ` AND jsonPayload.service_name IS NOT NULL`
-	queryString += ` AND jsonPayload.authentication_info.principal_email IS NOT NULL`
-	queryString += ") AS INT64)"
+	queryString := fmt.Sprintf(` WITH temptable AS (
+		SELECT *
+		FROM %s.%s
+		WHERE labels.trace_id=?
+ 	)
+ 	SELECT TO_JSON(t) as result FROM temptable as t
+	`, projectID, datasetQuery)
 	return makeQuery(client, id, queryString)
 }
+
+func diffResults(jsonPayloadInfo *audit.AuditLog, isHTTPService bool) string {
```

Comment:
I feel like writing our own diff is more complex, I saw your argument of ignoring too many fields is quite cumbersome, I checked our format of audit logs, I still feel that it is doable to ignore those fields like `metadata`, `status` and  `response`